Makefile

```diff
@@ -2,26 +2,43 @@ HOSTNAME=github.com
 NAMESPACE=iterative
 NAME=iterative
 VERSION=0.0.${shell date +%s}+development
-OS_ARCH=${shell go env GOOS}_${shell go env GOARCH}
-BINARY=terraform-provider-${NAME}
-INSTALL_PATH=~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${OS_ARCH}
+GOOS=${shell go env GOOS}
+GOARCH=${shell go env GOARCH}
+TF_PLUGIN_INSTALL_PATH=~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${GOOS}_${GOARCH}
+TPI_PATH ?= $(shell pwd)
+LEO_PATH ?= $(shell pwd)/cmd/leo
+GO_LINK_FLAGS ?= -s -w -X terraform-provider-iterative/iterative/utils.Version=${VERSION}
 
 default: build
 
+# TODO: Should this be renamed to something like build-check ?
+.PHONY: build
 build:
 	go build ./...
 
+.PHONY: leo
+leo:
+	CGO_ENABLED=0 \
+    	go build -ldflags="$(GO_LINK_FLAGS)" \
+    	-o $(shell pwd)/leo \
+    	$(LEO_PATH)
+
+.PHONY: install_tpi
```

Comment:
Are `.PHONY` targets needed after the “unholiness” was removed?

Reply:
PHONY are correct to add either way, otherwise dropping in a build file will cause a `make build` to be no-op and that's not what we want. `Makefile` the way it is used here (and any non classical-c-project way) is usually meant to ALWAYS execute the targets you run (assuming developer expectations here of course)

Reply:
Particularly the `leo` target 🤦🏼 👍🏼 🙏🏼